packages/core/workers/src/Worker.js

```diff
@@ -143,14 +145,19 @@ export default class Worker extends EventEmitter {
     let idx = this.callId++;
     this.calls.set(idx, call);
 
-    this.send({
-      type: 'request',
-      idx: idx,
-      child: this.id,
-      handle: call.handle,
-      method: call.method,
-      args: call.args,
-    });
+    (this.ready || call.skipReadyCheck === true
+      ? Promise.resolve()
```

Comment:
Maybe we can avoid the extra promise here in the common case?

Reply:
Done